File: source/python/566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, mat: list[list[int]], r: int, c: int) -> list[list[int]]:
        rows = len(mat)
        cols = len(mat[0])

        if rows * cols != r * c:
            return mat

        result = [[0] * c for _ in range(r)]

        for i in range(rows * cols):
            result[i // c][i % c] = mat[i // cols][i % cols]

        return result

File: source/typescript/566. Reshape the Matrix.ts
function matrixReshape(mat: number[][], r: number, c: number): number[][] {
    const rows = mat.length;
    const cols = mat[0].length;

    if (rows * cols != r * c)
        return mat;

    const result: number[][] = new Array(r).fill(0).map(() => new Array(c).fill(0));

    for (let i = 0; i < rows * cols; i++) {
        result[Math.floor(i / c)][i % c] = mat[Math.floor(i / cols)][i % cols];
    }

    return result;
};